projects/packages/videopress/src/client/block-editor/hooks/use-media-token.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import getMediaToken from '../../lib/get-media-token';
+import { MediaTokenScopeProps } from '../../lib/get-media-token/types';
+import { VideoGUID } from '../blocks/video/types';
+
+type UseMediaTokenArgs = {
+	guid?: VideoGUID;
+	isPrivate: boolean;
+	tokenType: MediaTokenScopeProps;
+};
+/**
+ * Hook to get a media token.
+ *
+ * @param {UseMediaTokenArgs} args - Hook arguments object.
+ * @returns {string} The media token.
+ */
+const useMediaToken = ( { guid, isPrivate, tokenType }: UseMediaTokenArgs ): string | null => {
+	const [ token, setToken ] = useState< string | null >( null );
+
+	useEffect( () => {
+		if ( ! isPrivate ) {
+			return setToken( null );
+		}
+		getMediaToken( tokenType, { id, guid } ).then( tokenData => {
```

Comment:
Looks like the `id` parameter passed here is missing. In fact, I don't see it referenced in the previous code ([reference](https://github.com/Automattic/jetpack/pull/29024/files#diff-884251507b133a31140b452f625b28062a669ef32271f13f4fb832beb3a82460L151)). Maybe we could remove it, wdyt?

Reply:
Ah yeah, the id param was just recently removed. Which actually caused some merge conflicts with this branch. I thought I took out all the usages but missed this one.